src/Json.Schema/JsonSchema.cs

```diff
@@ -691,5 +701,25 @@ public bool Equals(JsonSchema other)
         {
             return !(left == right);
         }
+
+        private static bool DefaultsAreEqual(object left, object right)
+        {
+            if ((left == null) != (right == null)) { return false; }
+
+            if (left == null) { return true; }
+
+            Type leftType = left.GetType(),
+                 rightType = right.GetType();
+            if (leftType != rightType) { return false; }
+
+            string leftString = left as String;
+            if (leftString != null) { return leftString.Equals(right as string, StringComparison.Ordinal); }
+
+            if (left is bool) { return (bool)left == (bool)right; }
+            if (left is long) { return (long)left == (long)right; }
+            if (left is double) { return (double)left == (double)right; }
+
+            return left == right; // Not good. How do we deal with other types?
```

Comment:
>Not [](start = 37, length = 3)

I'm open to suggestion here. Maybe `left.Equals(right)` is better, because at least it doesn't fail for two distinct objects that are value equal.

Reply:
Why aren't you simply using the Object.Equals implementation entirely for this? I see via reference source that it appears to shell out to some native code for the implementation. but it should handle all this type consideration/unboxing to do its thing. All this code used to be authored as C#, which I've reviewed in past contexts.

---
In reply to: [244400084](https://github.com/Microsoft/jschema/pull/60#discussion_r244400084) [](ancestors = 244400084)

Reply:
Thanks for that! It didn't even occur to me. Yes, after removing this function and instead calling `Object.Equals(Default, other.Default)`, all the tests still pass.

---
In reply to: [244415654](https://github.com/Microsoft/jschema/pull/60#discussion_r244415654) [](ancestors = 244415654,244400084)